projects/v2/src/app/modules/playground/playground.component.ts

```diff
@@ -267,14 +267,26 @@ export class PlaygroundComponent implements OnInit, AfterViewInit {
    * Read the google sheet link and upload
    */
   upload() {
-    const data = this.checkLinkFormat(this.linkFormControl.value);
-    const sheet = JSON.parse(JSON.stringify(this.currentSheet));
-    sheet.gid = data.gid;
-    sheet.sheetId = data.sheetID;
-    this.tabIndex = 0;
-    sheet.config.height = 1400;
-    this.store.dispatch(new FetchSheetData(sheet));
-    this.ga.eventEmitter('playground_upload', GaCategory.PLAYGROUND, 'Upload Playground Sheet', GaAction.CLICK, sheet.sheetId);
+    if (this.linkFormControl.value.endsWith('.csv')) {
+      const sheet = JSON.parse(JSON.stringify(this.currentSheet));
+      sheet.gid = '0';
+      sheet.sheetId = '0';
+      this.tabIndex = 0;
+      sheet.config.height = 1400;
+      this.store.dispatch(new FetchSheetDataFromCSV(sheet, this.linkFormControl.value));
+      this.ga.eventEmitter('playground_upload', GaCategory.PLAYGROUND, 'Upload Playground Sheet through CSV file',
```

Comment:
Also, need to support CSVs in the organ dropdown and compare functionality, not just playground.

Reply:
Got it